sql-plugin/src/main/scala/com/nvidia/spark/rapids/AlluxioUtils.scala

```diff
@@ -289,48 +289,90 @@ object AlluxioUtils extends Logging {
     }
 
     if (replaceFunc.isDefined) {
-      def isDynamicPruningFilter(e: Expression): Boolean =
-        e.find(_.isInstanceOf[PlanExpression[_]]).isDefined
-
-      val partitionDirs = relation.location.listFiles(
-        partitionFilters.filterNot(isDynamicPruningFilter), dataFilters)
-
-      // replace all of input files
-      val inputFiles: Seq[Path] = partitionDirs.flatMap(partitionDir => {
-        partitionDir.files.map(f => replaceFunc.get(f.getPath))
-      })
+      def replacePathsInPartitionSpec(spec: PartitionSpec): PartitionSpec = {
+        val partitionsWithPathsReplaced = spec.partitions.map { p =>
+          val replacedPath = replaceFunc.get(p.path)
+          org.apache.spark.sql.execution.datasources.PartitionPath(p.values, replacedPath)
+        }
+        PartitionSpec(spec.partitionColumns, partitionsWithPathsReplaced)
+      }
 
-      // replace all of rootPaths which are already unique
-      val rootPaths = relation.location.rootPaths.map(replaceFunc.get)
+      def createNewFileIndexWithPathsReplaced(
+          spec: PartitionSpec,
+          rootPaths: Seq[Path]): InMemoryFileIndex = {
+        val specAdjusted = replacePathsInPartitionSpec(spec)
+        val replacedPaths = rootPaths.map(p => replaceFunc.get(p))
```

Comment:
```suggestion
        val replacedPaths = rootPaths.map(replaceFunc.get)
```

I kind of wish we would `.get` the function at the top and store it in `replaceFunc` then change the original variable to `maybeReplaceFunc`.